bundles/core/org.eclipse.smarthome.core.library.test/src/test/java/org/eclipse/smarthome/core/library/types/LocationTypeTest.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.library.types;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.eclipse.smarthome.core.library.items.LocationItem;
+import org.junit.Test;
+
+/**
+ * @author Gaël L'hopital
+ */
+public class LocationTypeTest {
+
+    @Test
+    public void testDistance() {
+        PointType pointParis = new PointType("48.8566140,2.3522219");
+
+        
+        assertEquals(2.3522219, pointParis.getLongitude().doubleValue(), 0.0000001);
+        assertEquals(48.856614, pointParis.getLatitude().doubleValue(), 0.0000001);
+        
+        PointType pointBerlin = new PointType("52.5200066,13.4049540");
+
+        LocationItem locationParis = new LocationItem("paris");
+        locationParis.setState(pointParis);
+        LocationItem locationBerlin = new LocationItem("berlin");
+        locationBerlin.setState(pointBerlin);
+
+        DecimalType distance = locationParis.distanceFrom(pointParis);
+        assertEquals(0, distance.intValue());
+
+        double parisBerlin = locationParis.distanceFrom(pointBerlin).doubleValue();
+        assertEquals(878400, 50, parisBerlin);
```

Comment:
Ensure that distance calculation provided by LocationItem meets real measure provided by Google Maps